packages/components/src/components/editor/condition-builder/conditionBuilder.stories.tsx

```diff
@@ -0,0 +1,110 @@
+      <>
+        <ConditionBuilderProvider
+          generateConditionString={GenerateConditionString}
+          logicalOperatorOptions={LogicalOperatorOptions}
+          typeOptions={TypeOptions}
+        >
+          <ConditionBuilder onChange={setCondition} />
+        </ConditionBuilderProvider>
+        <pre
+          style={{ border: 'var(--basic-border)', maxHeight: '80px', padding: 'var(--input-padding)', borderRadius: 'var(--border-r2)' }}
+        >
+          {condition}
+        </pre>
+      </>
+    );
+  }
+};
+
+export const TypeOptions: TypeOperator = {
+  'equal to': '==',
+  'not equal to': '!=',
+  'is true': 'isTrue',
+  'is false': 'isFalse',
+  'is empty': 'isEmpty',
+  'is not empty': 'isNotEmpty',
+  'less than': '<',
+  'greater than': '>',
+  'less or equal to': '<=',
+  'greater or equal to': '>='
+};
+
+export const LogicalOperatorOptions: LogicOperator = {
+  'and': '&&',
+  'or': '||'
+};
+
+export const GenerateConditionString = (conditionMode: ConditionMode, conditionGroups: ConditionGroup[]) => {
+  if (conditionMode === 'always-true') {
+    return 'true';
+  }
+  if (conditionMode === 'always-false') {
+    return 'false';
+  }
+  const wrapInQuotesIfNeeded = (arg: string) => {
+    return arg.includes('.') || arg.startsWith('in') || arg.startsWith('out') ? arg : `"${arg}"`;
+  };
+
+  const groupStrings = conditionGroups.map((group, index) => {
+    if (conditionMode === 'basic-condition' && index > 0) return '';
+    const conditions = group.conditions
+      .map((con, conditionIndex) => {
+        const formattedArg1 = wrapInQuotesIfNeeded(con.argument1);
+        const formattedArg2 = wrapInQuotesIfNeeded(con.argument2);
+        const logicalOp = getLogicalOperator(conditionIndex, group.conditions.length, con.logicalOperator);
+        return formatCondition(con, formattedArg1, formattedArg2, logicalOp);
+      })
+      .join('');
+
+    const logicGroupOp = index < conditionGroups.length - 1 ? group.logicalOperator.toLowerCase() : '';
+    return conditionGroups.length === 1 || conditionMode === 'basic-condition'
+      ? conditions
+      : `(${conditions})${logicGroupOp ? ` ${logicGroupOp} ` : ''}`;
+  });
+
+  return groupStrings.join('');
+};
+
+const getLogicalOperator = (index: number, totalConditions: number, logicalOperator: LogicalOperator) => {
+  return index < totalConditions - 1 ? ` ${LogicalOperatorOptions[logicalOperator].toLowerCase()} ` : '';
+};
+
+const formatCondition = (con: Condition, formattedArg1: string, formattedArg2: string, logicalOp: string) => {
+  switch (con.operator) {
+    case 'is true':
+      return `${formattedArg1}${logicalOp}`;
+    case 'is false':
+      return `!${formattedArg1}${logicalOp}`;
+    case 'is empty':
+      return `${formattedArg1}.isEmpty()${logicalOp}`;
+    case 'is not empty':
+      return `!${formattedArg1}.isEmpty()${logicalOp}`;
+    default:
+      return `${formattedArg1} ${TypeOptions[con.operator]} ${formattedArg2}${logicalOp}`;
```

Comment:
Here you can see that as a consequence that ConditionType directly uses keyof TypeOperator and not the actual value of the attribute, I need to search for the correct value in the implementation (if it's unclear, we can discuss it in person)

Reply:
I'm not sure what you mean, but this is fine or not?
the con.operator is one of the keys of the TypeOperator, and the value for this key is the thing which is different and the thing we need to put here.